utils_nlp/models/transformers/named_entity_recognition.py

```diff
@@ -1,2 +1,430 @@
+)
+
+from utils_nlp.models.transformers.common import (
+    MAX_SEQ_LEN,
+    TOKENIZER_CLASS,
+    Transformer,
+    get_device,
+)
+
+TC_MODEL_CLASS = {k: BertForTokenClassification for k in BERT_PRETRAINED_MODEL_ARCHIVE_MAP}
+
+
+class TokenClassificationProcessor():
+    """
+    Process raw dataset for training and testing, and post-process prediction
+    for token classification.
+
+    Args:
+        model_name (str, optional): The pretained model name.
+            Defaults to "bert-base-cased".
+        to_lower (bool, optional): Lower case text input.
+            Defaults to False.
+        cache_dir (str, optional): The default folder for saving cache files.
+            Defaults to ".".
+    """
+
+    def __init__(self, model_name="bert-base-cased", to_lower=False, cache_dir="."):
+        self.model_name = model_name
+        self.to_lower = to_lower
+        self.cache_dir = cache_dir
+        self.tokenizer = TOKENIZER_CLASS[model_name].from_pretrained(
+            model_name, do_lower_case=to_lower, cache_dir=cache_dir
+        )
+
+    @staticmethod
+    def get_inputs(batch, model_name, train_mode=True):
+        """
+        Produce a dictional object for model training or prediction.
+
+        Args:
+            model_name (str): The pretained model name.
+            train_mode (bool, optional): Whether it's for model training. Set it to False if
+                it's for testing and it won't have the 'labels' data field.
+                Defaults to True, for model training.
+        
+        Returns:
+            A dictional object contains all needed information for training or testing.
+        """
+
+        if model_name.split("-")[0] not in ["bert"]:
+            raise ValueError("Model not supported: {}".format(model_name))
+    
+        if train_mode:
+            return {"input_ids": batch[0], "attention_mask": batch[1], "labels": batch[3]}
+        else:
+            return {"input_ids": batch[0], "attention_mask": batch[1]}
+
+
+    @staticmethod
+    def create_label_map(label_lists, trailing_piece_tag="X"):
+        """
+        Create a dictional object to map a label (str) to an ID (int).
+
+        Args:
+            label_lists (list): A list of label lists. Each element is a list of labels
+                which presents class of each token.
+            trailing_piece_tag (str, optional): Tag used to label trailing word pieces.
+                Defaults to "X".
+            
+        Returns:
+            A dictional object to map a label (str) to an ID (int). 
+        """
+
+        label_set = set()
+        for labels in label_lists:
+            label_set.update(labels)
+
+        label_map = {label: i for i, label in enumerate(label_set)}
+
+        if trailing_piece_tag not in label_set:
+            label_map[trailing_piece_tag] = len(label_set)
+        return label_map
+
+
+    def preprocessForBERT(self, text, max_len=MAX_SEQ_LEN, labels=None, label_map=None, trailing_piece_tag="X"):
```

Comment:
I think we can just call this method "preprocess". Ideally, it should return the correct format corresponding to the model_name of the processor.

Reply:
I prefer to use "preprocessForBERT" just for future updates. I believe different Transformer models, the inputs for the models are different. I think it's better to write different functions for different models such that we can easily distinguish them and control what format of data we should produce. We can discuss more on this topic during our Dataset design meeting.

Reply:
OK. For sequence classification and qa we only had one, because there are a lot over lapping of preprocessing code for different transformers. We can definitely further discuss what's the best way. 
For now, can we change it to preprocess_for_bert for consistent method name format?

Reply:
Done.